client/component.go

```diff
@@ -1,23 +1,107 @@
+// SPDX-License-Identifier: Apache-2.0
+
+// Copyright 2023 PANTHEON.tech
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
 package client
 
-import "go.pantheon.tech/stonework/proto/cnfreg"
+import (
+	"context"
+	"fmt"
+
+	"go.ligato.io/vpp-agent/v3/cmd/agentctl/api/types"
+	"go.ligato.io/vpp-agent/v3/cmd/agentctl/client"
+
+	"go.ligato.io/vpp-agent/v3/proto/ligato/kvscheduler"
+	"go.pantheon.tech/stonework/plugins/cnfreg"
+	cnfregpb "go.pantheon.tech/stonework/proto/cnfreg"
+)
+
+type ComponentMode int32
+
+const (
+	// Foreign means the component is not managed by StoneWork
+	ComponentForeign ComponentMode = iota
+
+	// StoneworkModule means the component is a StoneWork module managed by StoneWork
+	ComponentStoneworkModule
+
+	// Stonework means the component is a StoneWork instance
+	ComponentStonework
+)
 
 // Component is a component of StoneWork. It can be StoneWork instance itself,
 // a CNF connected to it or other Ligato service in connected to StoneWork.
 type Component interface {
-	Name() string
-	Mode() cnfreg.CnfMode
+	GetName() string
+	GetMode() ComponentMode
+	GetInfo() *cnfreg.Info
+	GetMetadata() map[string]string
+	SchedulerValues() ([]*kvscheduler.BaseValueStatus, error)
```

Comment:
I do not think we want to export a low-level method like this, which returns status for scheduler values. StoneWork user has not idea what it is.. and more importantly, will not know what to do with the returned values.. try to think from perspective of user and only export the processed data that user wants to know about _a component_. The retrieval of scheduler values can still happen in the background, but it should not be transparently passed to our users.

Reply:
Replaced with a method that directly outputs only the counts to the user.